File: lib.net/src/main/java/com/laka/libnet/rx/ApiService.java
package com.laka.libnet.rx;

import com.laka.libnet.rx.response.ResponseInfo;

import java.util.HashMap;
import java.util.Map;

import io.reactivex.Observable;
import retrofit2.http.FieldMap;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.QueryMap;
import retrofit2.http.Url;

/**
 * @Author:summer
 * @Date:2019/7/19
 * @Description:
 */
public interface ApiService {

    @GET("s?wd=今日新鲜事&tn=SE_PclogoS_8whnvm25&sa=ire_dl_gh_logo&rsv_dl=igh_logo_pcs")
    Observable<String> getUserInfo(@QueryMap HashMap<String,String> params);

    @POST()
    @FormUrlEncoded
    Observable<ResponseInfo> postRequest(@Url() String url, @FieldMap Map<String, String> params);

    @POST()
    Observable<ResponseInfo> postRequest(@Url() String url);

    @GET()
    @FormUrlEncoded
    Observable<ResponseInfo> getRequest(@Url() String url, @QueryMap Map<String, String> params);

    @GET()
    Observable<ResponseInfo> getRequest(@Url() String url);

}

File: lib.net/src/main/java/com/laka/libnet/rx/ApiService.kt
package com.laka.libnet.rx

import com.laka.libnet.rx.response.ResponseInfo
import io.reactivex.Observable
import retrofit2.http.*
import java.util.HashMap

interface ApiService {


    @GET("s?wd=今日新鲜事&tn=SE_PclogoS_8whnvm25&sa=ire_dl_gh_logo&rsv_dl=igh_logo_pcs")
    fun getUserInfo(@QueryMap params: HashMap<String, String>): Observable<String>

    @POST
    @FormUrlEncoded
    fun postRequest(@Url url: String, @FieldMap params: Map<String, String>): Observable<ResponseInfo>

    @POST
    fun postRequest(@Url url: String): Observable<ResponseInfo>

    @GET
    @FormUrlEncoded
    fun getRequest(@Url url: String, @QueryMap params: Map<String, String>): Observable<ResponseInfo>

    @GET
    fun getRequest(@Url url: String): Observable<ResponseInfo>

    @GET
    suspend fun getUserInfo(@Url url: String): ResponseInfo

}